packages/typespec-azure-resource-manager/lib/operations.tsp

```diff
@@ -108,9 +108,32 @@ op ArmResourceRead<
   ErrorResponse
 >;
 
+/**
+ * A resource HEAD operation
+ * @template Resource the resource being checked
+ * @template BaseParameters Optional. Allows overriding the operation parameters
+ * @template Parameters Optional. Additional parameters after the path parameters
+ * @template Response Optional. The success response for the read operation
+ * @template Error Optional. The error response, if non-standard.
+ */
+@autoRoute
+@doc("Check a {name} via HEAD operation", Resource)
+@head
+op ArmResourceHead<
```

Comment:
Should this be `ArmResourceExists`?

Reply:
I was thinking about the same thing but was concerned with CheckNameExist operations. But hearing second on it, I will update it and bring to ARM for final review.

Reply:
I think existence check is one of the use cases for the endpoint. It is not the only one. We have chosen to provide a client library convention that uses it for existence checks. But that is more of a client library concern, no?

Reply:
This is scoped to ARM which has its owe standards such as name and response codes. Client Lib may provide a template that is more generic and would not conflict with this one similar to other Life cycle operations.